app/schema/portal/cdo/dog-details.js

```diff
@@ -20,56 +20,72 @@ const parseCdoIssueDate = (value) => {
   return null
 }
 
-const validateDate = (value, helpers) => {
-  if (value === '--') {
-    return helpers.message('Enter a CDO issue date')
+const calculateExpiryDate = (value) => {
+  const dateString = `${value.year}-${value.month}-${value.day}`
+
+  return addMonths(parseCdoIssueDate(dateString), 2)
+}
+
+const validateIssueDate = (value, helpers) => {
+  const { day, month, year } = value
+  const dateComponents = { day, month, year }
+  const invalidComponents = []
+
+  for (const key in dateComponents) {
+    if (!dateComponents[key]) {
+      invalidComponents.push(key)
+    }
   }
 
-  const date = parseCdoIssueDate(value)
+  if (invalidComponents.length === 0) {
+    const dateString = `${year}-${month}-${day}`
+    const date = parseCdoIssueDate(dateString)
+
+    if (!date) {
+      return helpers.message('Enter a real date', { path: ['cdoIssued', ['day', 'month', 'year']] })
+    }
+
+    if (isFuture(date)) {
+      return helpers.message('Enter a date that is in the past', { path: ['cdoIssued', ['day', 'month', 'year']] })
+    }
 
-  if (date) {
     return date
   }
 
-  return helpers.message('CDO issue date must be a real date')
-}
-
-const validateIssueDate = (value, helpers) => {
-  if (typeof value === 'object' && isFuture(value)) {
-    return helpers.message('CDO issue date must not be in the future')
+  if (invalidComponents.length === 3) {
+    return helpers.message('Enter a CDO issue date', { path: ['cdoIssued', ['day', 'month', 'year']] })
   }
 
-  return value
+  const errorMessage = `A CDO issue date must include a ${invalidComponents.join(' and ')}`
+
+  return helpers.message(errorMessage, { path: ['cdoIssued', invalidComponents] })
 }
 
 const dogDetailsSchema = Joi.object({
   breed: Joi.string().trim().required().messages({
     '*': 'Breed type is required'
   }),
-  name: Joi.string().trim().allow('').allow(null).optional(),
-  cdoIssued: Joi.string().required().messages({
-    'string.base': 'Enter a CDO issue date',
-    'string.empty': 'Enter a CDO issue date'
-  }).custom(validateDate).custom(validateIssueDate),
+  name: Joi.string().trim().max(32).allow('').allow(null).optional().messages({
+    'string.max': 'Dog name must be {#limit} characters or fewer'
```

Comment:
Not an issue but we've previously used the syntax '<field> must be no more than {x} characters'